evalml/pipelines/components/estimators/regressors/multiseries_time_series_baseline_regressor.py

```diff
@@ -0,0 +1,112 @@
+    """{}"""
+    model_family = ModelFamily.BASELINE
+    """ModelFamily.BASELINE"""
+    is_multiseries = True
+    supported_problem_types = [
+        ProblemTypes.TIME_SERIES_REGRESSION,
+    ]
+    """[
+        ProblemTypes.TIME_SERIES_REGRESSION,
+    ]"""
+
+    def __init__(self, gap=1, forecast_horizon=1, random_seed=0, **kwargs):
+        self._prediction_value = None
+        self.start_delay = forecast_horizon + gap
+        self._num_features = None
+
+        if gap < 0:
+            raise ValueError(
+                f"gap value must be a positive integer. {gap} was provided.",
+            )
+
+        parameters = {"gap": gap, "forecast_horizon": forecast_horizon}
+        parameters.update(kwargs)
+        super().__init__(
+            parameters=parameters,
+            component_obj=None,
+            random_seed=random_seed,
+        )
+
+    def fit(self, X, y=None):
+        """Fits multiseries time series baseline regressor to data.
+
+        Args:
+            X (pd.DataFrame): The input training data of shape [n_samples, n_features * n_series].
+            y (pd.DataFrame): The target training data of shape [n_samples, n_features * n_series].
+
+        Returns:
+            self
+
+        Raises:
+            ValueError: If input y is None.
+        """
+        if y is None:
+            raise ValueError(
+                "Cannot train Multiseries Time Series Baseline Regressor if y is None",
+            )
+        if isinstance(y, pd.Series):
+            raise ValueError(
+                "y must be a DataFrame with multiple columns for Multiseries Time Series Baseline Regressor",
+            )
+        self._target_column_names = list(y.columns)
+        self._num_features = X.shape[1]
+
+        return self
+
+    def predict(self, X):
+        """Make predictions using fitted multiseries time series baseline regressor.
+
+        Args:
+            X (pd.DataFrame): Data of shape [n_samples, n_features].
+
+        Returns:
+            pd.DataFrame: Predicted values.
+
+        Raises:
+            ValueError: If the lagged columns are not present in X.
+        """
+        X = infer_feature_types(X)
+        feature_names = [
+            f"{col}_delay_{self.start_delay}" for col in self._target_column_names
+        ]
+        if not set(feature_names).issubset(set(X.columns)):
+            raise ValueError(
+                "Multiseries Time Series Baseline Regressor is meant to be used in a pipeline with "
+                "a Time Series Featurizer",
+            )
+        return X.ww[feature_names]
+
+    @property
+    def feature_importance(self):
+        """Returns importance associated with each feature.
+
+        Since baseline estimators do not use input features to calculate predictions, returns an array of zeroes.
+
+        Returns:
+            np.ndarray (float): An array of zeroes.
+        """
+        importance = np.array([0] * self._num_features)
+        return importance
```

Comment:
Probably another nit, but if you're calling out all Baseline Estimators...is it worth putting together a story to add a BaselineEstimator class to the inheritance chain and have them all inherit this prop def?

Reply:
Filed #4255